go/host/host.go

```diff
@@ -504,29 +504,23 @@ func (h *host) processL1Block(block *types.Block, isLatestBlock bool) error {
 	}
 
 	// If we're the sequencer, and we're processing the latest block, we produce, publish and distribute a new rollup.
-	if !h.isSequencer || !isLatestBlock {
+	if !h.isSequencer {
 		return nil
 	}
 
 	if h.genesisRequired {
-		err := h.initialiseProtocol(block)
-		if err != nil {
+		if err = h.initialiseProtocol(block); err != nil {
 			h.logger.Crit("Could not initialise protocol.", log.ErrKey, err)
 		}
 		h.genesisRequired = false
 		return nil // nothing further to process since network had no genesis
 	}
 
-	rollup, err := h.enclaveClient.ProduceRollup()
-	if err != nil {
-		return fmt.Errorf("could not produce rollup. Cause: %w", err)
-	}
-
-	if rollup.Header != nil {
+	if result.ProducedRollup != nil && result.ProducedRollup.Header != nil {
```

Comment:
The first check works for the in-mem sims, and the second for the full-network sims. Insane.